sycl/test-e2e/AddressSanitizer/common/kernel-debug.cpp

```diff
@@ -0,0 +1,17 @@
+// REQUIRES: linux, cpu
+// RUN: %{build} %device_sanitizer_flags -O2 -g -o %t
+// RUN: env SYCL_PREFER_UR=1 UR_LAYER_ASAN_OPTIONS=debug:1 %{run} %t 2>&1 | FileCheck --check-prefixes CHECK-DEBUG %s
```

Comment:
How does it work without `UR_ENABLE_LAYERS=UR_LAYER_ASAN`?

Reply:
Thanks for reviewing this PR.

Reply:
`UR_ENABLE_LAYERS=UR_LAYER_ASAN` is used to forcedly enable ASan Layer in UR loader, which is the runtime support for device sanitizer.
We enable UR_LAYER_ASAN automatically by checking kernel image property.
But some tests don't have any kernel, so I have to enable UR_LAYER_ASAN manually.
Since the user is unlikely to need enable device ASan without any kernels, I think this flag can be used just for testing.